### PR TITLE
Introduce `GetJsonObjectOptions` in `getJSONObject` Java API

### PR DESCRIPTION
**Pull request by SurajAralihalli**
## Description
Resolves [10219](https://github.com/NVIDIA/spark-rapids/issues/10219)

This PR introduces a new class named `GetJsonObjectOptions` that holds the configurations to control the behavior of the underlying `cudf::get_json_object` function. It incorporates this new class into the `getJSONObject` JAVA API as an additional argument but also keeps the previous API to maintain backwards compatibility.  It also includes a test case, `testGetJSONObjectWithSingleQuotes`, validating the behavior of `getJSONObject` when single quotes are enabled.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by revans2 — APPROVED** [review]
Looks great

**Comment by revans2** [comment]
@karthikeyann it really is about JNI limitations.  We have a few choices for what we could do.

We could have a java class that creates an instance of a C++ object and holds a reference to it. Then anything that interacts with the object needs to call into JNI to change things. It also has the possibility of being leaked if the java code does not close it, or if we don't put in a `finalizer` to make sure that the data is released when GC happens. Neither is very user friendly from a java standpoint. 

We could have a pure java class that holds the values, and we could pass an instance of that class into JNI. But it is very slow to call back into java from JNI. So if we know that all of the values are going to be used, we typically just pull it apart in java and send them down as separate parameters so the performance is not too bad.

**Review by revans2 — APPROVED** [review]

File: java/src/main/native/src/ColumnViewJni.cpp [hunk 1]
```diff
@@ -2448,7 +2448,11 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getJSONObject(JNIEnv *env
     cudf::column_view *n_column_view = reinterpret_cast<cudf::column_view *>(j_view_handle);
     cudf::strings_column_view n_strings_col_view(*n_column_view);
     cudf::string_scalar *n_scalar_path = reinterpret_cast<cudf::string_scalar *>(j_scalar_handle);
-    return release_as_jlong(cudf::get_json_object(n_strings_col_view, *n_scalar_path));
+    auto options = cudf::get_json_object_options{};
+    options.set_allow_single_quotes(allow_single_quotes);
+    options.set_strip_quotes_from_single_strings(strip_quotes_from_single_strings);
+    options.set_missing_fields_as_nulls(missing_fields_as_nulls);
```
Comment (mythrocks) [inline java/src/main/native/src/ColumnViewJni.cpp:2454]: I do wish `cudf::get_json_object_options` had a fluent interface for the setters.  But that's not the fault of this current change.

**Review by mythrocks — APPROVED** [review]

File: java/src/main/native/src/ColumnViewJni.cpp (hunk 1 above)
Comment (SurajAralihalli) [reply java/src/main/native/src/ColumnViewJni.cpp:2454]: Right, thanks for reviewing!

**Review by karthikeyann — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Pull request merged**